scipy/optimize/_differentialevolution.py

```diff
@@ -962,14 +962,7 @@ def __enter__(self):
         return self
 
     def __exit__(self, *args):
-        # to make sure resources are closed down
-        self._mapwrapper.close()
-        self._mapwrapper.terminate()
-
-    def __del__(self):
```

Comment:
At the moment it's probably unlikely that `DifferentialEvolutionSolver` is called anywhere outside `differential_evolution` which would close down the resources properly. If that's done then `__del__` here may not be necessary.
However, I'm not sure getting rid of the `__del__` in `MapWrapper` is a good idea.

Reply:
From the documentation of `Pool.terminate`:
> When the pool object is garbage collected terminate() will be called immediately.

So `terminate` will be called no matter what, and the `__del__` method in `MapWrapper` is unnecessary.